src/plugins/event-plugins/__integ__/JsErrorPlugin.spec.ts

```diff
@@ -0,0 +1,100 @@
+        page
+    }) => {
+        await page.goto('/js_error_event.html');
+
+        await page.waitForTimeout(300);
+        await page.click('#triggerTypeError');
+        await page.click('#dispatch');
+
+        await expect(page.locator('#request_body')).toContainText('BatchId');
+
+        const requestBodyText = await page
+            .locator('#request_body')
+            .textContent();
+        const events = JSON.parse(requestBodyText || '{}').RumEvents.filter(
+            (e: any) => e.type === JS_ERROR_EVENT_TYPE
+        );
+        const eventType = events[0].type;
+        const eventDetails = JSON.parse(events[0].details);
+
+        expect(eventType).toBe(JS_ERROR_EVENT_TYPE);
+        expect(eventDetails.type).toContain('TypeError');
+        expect(eventDetails.message).toMatch(/(undefined|null)/);
+        expect(eventDetails.filename).toMatch(/js_error_event.html/);
+        expect(typeof eventDetails.lineno).toBe('number');
+        expect(typeof eventDetails.colno).toBe('number');
+    });
+
+    test('when a promise rejection is thrown then name is recorded', async ({
+        page
+    }) => {
+        await page.goto('/js_error_event.html');
+
+        await page.waitForTimeout(300);
+        await page.click('#uncaughtPromiseRejection');
+        await page.click('#dispatch');
+
+        await expect(page.locator('#request_body')).toContainText('BatchId');
+
+        const requestBodyText = await page
+            .locator('#request_body')
+            .textContent();
+        const json = removeUnwantedEvents(JSON.parse(requestBodyText || '{}'));
+        const eventType = json.RumEvents[0].type;
+        const eventDetails = JSON.parse(json.RumEvents[0].details);
+
+        expect(eventType).toBe(JS_ERROR_EVENT_TYPE);
+        expect(eventDetails.type).toContain('unhandledrejection');
+        expect(eventDetails.message).toMatch(/promise is rejected/);
+    });
+
+    test('when the application records a caught error then the plugin records the error', async ({
+        page
+    }) => {
+        await page.goto('/js_error_event.html');
+
+        await page.waitForTimeout(300);
+        await page.click('#recordCaughtError');
+        await page.click('#dispatch');
+
+        await expect(page.locator('#request_body')).toContainText('BatchId');
+
+        const requestBodyText = await page
+            .locator('#request_body')
+            .textContent();
+        const events = JSON.parse(requestBodyText || '{}').RumEvents.filter(
+            (e: any) => e.type === JS_ERROR_EVENT_TYPE
+        );
+
+        const eventType = events[0].type;
+        const eventDetails = JSON.parse(events[0].details);
+
+        expect(events.length).toBe(1);
+        expect(eventType).toBe(JS_ERROR_EVENT_TYPE);
+        expect(eventDetails.type).toContain('Error');
+        expect(eventDetails.message).toContain('My error message');
+    });
+});
```

Comment:
question: Seems like we're missing some tests, was this intentional?

1. stack trace is recorded by default
2. when a string is thrown then name and message are recorded
3. when the application records a caught error then the plugin records the error
4. when ignore function matches error then the plugin does not record the error
5. when error invoked with record method then the plugin records the error